cms/search/utils.py

```diff
@@ -0,0 +1,70 @@
+from typing import TYPE_CHECKING
+
+from django.apps import apps
+from django.utils.encoding import force_str
+from wagtail.rich_text import get_text_for_indexing
+
+from cms.release_calendar.enums import ReleaseStatus
+
+if TYPE_CHECKING:
+    from wagtail.models import Page
+
+
+def build_standard_resource_dict(page: "Page") -> dict:
+    """Returns a dict with the standard resource fields.
+    This covers the non-release case (and also forms the base of the release case).
+    """
+    return {
+        "uri": page.url_path.removeprefix("/home"),
+        "content_type": page.search_index_content_type,
+        "release_date": (page.release_date.isoformat() if getattr(page, "release_date", None) else None),
+        "summary": get_text_for_indexing(force_str(page.summary)),
+        "title": page.title,
+        "topics": getattr(page, "topic_ids", []),
+        "language": force_str(page.locale.get_display_name()),
```

Comment:
```suggestion
        "language": force_str(get_locales_display_names().get(page.locale_id)),
```

with `from wagtail.coreutils import get_locales_display_names` - this makes use of the cached locales display names, and reduces the need to additional queries/joins on the locale table.

Reply:
Done!